src/olympia/versions/views.py

```diff
@@ -90,9 +89,12 @@ def update_info_redirect(request, version_id):
 
 # Should accept junk at the end for filename goodness.
 @non_atomic_requests
-def download_file(request, file_id, type=None):
-    file = get_object_or_404(File.objects, pk=file_id)
-    addon = get_object_or_404(Addon.with_unlisted, pk=file.version.addon_id)
+def download_file(request, file_id, type=None, file=None, addon=None):
```

Comment:
Can you use the occasion to rename the variable to `file_` to avoid the clash with the builtin `file` ?